File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/context/XmlElementContext_ViewTests.cpp
#include <gtest/gtest.h>
#include <memory>
#include "XmlElementContext_ViewModelTestEnvironment.h"
#include "datadriven/DataDrivenViewModelTestEnvironment.h"
#include "XmlElementContext_ViewModel.h"
#include "XmlElementContext_ViewModelTestEnvironmentImpl.h"
#include <string>

namespace context
{
  class XmlElementContext_ViewTests : public testing::Test
  {
  public:
    std::shared_ptr<XmlElementContext_ViewModel> sut;
    std::shared_ptr<XmlElementContext_ViewModelTestEnvironment> testEnvironment;
    virtual void BuildSut();
    std::string myxml = R"(<MyXML>
  <Inner></Inner>
</MyXML>)";
    virtual void given_myxml();
    std::string MyXML_Inner = R"(<MyXML>
  <Inner></Inner>
</MyXML>)";
    virtual void given_MyXML_Inner();
  protected:
    void SetUp() override;
  };
  void XmlElementContext_ViewTests::SetUp()
  {
    this->testEnvironment = std::make_shared<XmlElementContext_ViewModelTestEnvironmentImpl>();
    this->testEnvironment->Init();
  }
  TEST_F(XmlElementContext_ViewTests, Xml_Context_given_myxml_when_then_)
  {
    this->given_myxml();
    this->BuildSut();
  }
  TEST_F(XmlElementContext_ViewTests, Xml_Context_explicit_name_given_MyXML_Inner_when_then_)
  {
    this->given_MyXML_Inner();
    this->BuildSut();
  }
  void XmlElementContext_ViewTests::BuildSut()
  {
    this->sut = this->testEnvironment->BuildSut();
  }
  void XmlElementContext_ViewTests::given_myxml()
  {
    this->testEnvironment->SetXmlElementContext(this->myxml);
  }
  void XmlElementContext_ViewTests::given_MyXML_Inner()
  {
    this->testEnvironment->SetXmlElementContext(this->MyXML_Inner);
  }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/context/XmlElementContext_ViewTests.cs
/// <filename>
///     XmlElementContext_ViewTests.cs
/// </filename>
using context;
using datadriven;
using Microsoft.VisualStudio.TestTools.UnitTesting;

[Microsoft.VisualStudio.TestTools.UnitTesting.TestClass]
public class XmlElementContext_ViewTests
{
    private context.XmlElementContext_ViewModel sut;

    private context.XmlElementContext_ViewModelTestEnvironment testEnvironment;

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestInitialize]
    public void SetUp()
    {
        this.testEnvironment = new context.XmlElementContext_ViewModelTestEnvironmentImpl();
        this.testEnvironment.Init();
    }

    /*

    Scenario: Xml Context
      given: myxml
       when:
       then:
   */
    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void Xml_Context_given_myxml_when_then_()
    {
        this.given_myxml();
        this.BuildSut();
    }

    /*

    Scenario: Xml Context (explicit name)
      given: MyXML_Inner
       when:
       then:
   */
    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void Xml_Context_explicit_name_given_MyXML_Inner_when_then_()
    {
        this.given_MyXML_Inner();
        this.BuildSut();
    }

    protected virtual void BuildSut()
    {
        this.sut = this.testEnvironment.BuildSut();
    }

    // Given Helper Definitions
    private string myxml = @"<MyXML>
  <Inner></Inner>
</MyXML>";

    public virtual void given_myxml()
    {
        this.testEnvironment.SetXmlElementContext(this.myxml);
    }

    private string MyXML_Inner = @"<MyXML>
  <Inner></Inner>
</MyXML>";

    public virtual void given_MyXML_Inner()
    {
        this.testEnvironment.SetXmlElementContext(this.MyXML_Inner);
    }
}
